src/chatClient.cpp

```diff
@@ -320,6 +320,35 @@ bool Client::openDb(const std::string& sid)
                 ok = true;
                 KR_LOG_WARNING("Database version has been updated to %s", gDbSchemaVersionSuffix);
             }
+            else if (cachedVersionSuffix == "10" && (strcmp(gDbSchemaVersionSuffix, "11") == 0))
+            {
+                // Remove USER_ATTR_RSA_PUBKEY attr from cache
+                db.query("delete from userattrs where type = 64");
+
+                // Create temporary table and copy sendkeys content
+                db.query("CREATE TABLE tempkeys(chatid int64 not null, userid int64 not null, keyid int32 not null, key blob not null,ts int not null, UNIQUE(chatid, userid, keyid));");
+                db.query("INSERT INTO tempkeys(chatid, userid, keyid, key, ts) SELECT chatid, userid, keyid, key, ts FROM sendkeys");
+
+                // Close and re-open db again to avoid SQLITE_LOCKED
```

Comment:
I don't get this one. Why is it locked?

Reply:
https://ti-qa-archive.github.io/question/119597/sqlite-error-database-table-is-locked.html

The database file will lock when a write action is being  performed such as a insert, delete or update. You'll need to ensure that  you close and/or end the transaction of any insert/delete/update  queries to prevent a locking run-time error. It's also worth noting that  currently a try-catch around the query execution will not catch any SQL  error notice.

Reply:
No other thread should be reading from this DB. We are initializing it, even before anyone else can write or read from this DB, so it shouldn't be locked. Please, retest without the `close` + `open` and ensuring no other DB connections are opened.